odoo/models.py

```diff
@@ -3850,13 +3852,20 @@ def get_trans(field, old, new):
             if not field.copy:
                 continue
 
+            if field.inherited and field.related[0] in excluded_fields:
+                # inherited fields that are using a user-provided value
+                # must not copy translation but assume the translations are
+                # already handled on the new relation
+                continue
+
             if field.type == 'one2many':
```

Comment:
Similar problem here: what if a one2many field is given a value?
```python
line = sale_order.order_line[0].copy(...)
sale_order.copy({'order_line': [(6, 0, line.ids)]})
```
There should be no recursive copy in this case...